### PR TITLE
post: add zig build explained series

### PR DESCRIPTION
**Pull request by 1000copy**

### CODE REVIEW

**Comment by jiacai2050** [comment]
@1000copy CI failed. 😅

**Comment by 1000copy** [comment]
> @1000copy CI failed. 😅

you can try it again:)

File: content/post/2023-12-24-zig-build-explained-part1.md [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+    const std = @import("std");
+    pub fn build(b: *std.build.Builder) void {
+        const exe = b.addExecutable(.{.name = "fresh",.root_source_file = .{ .path = "src/main.zig" },});
+        const compile_step = b.step("compile", "Compiles src/main.zig");
+        compile_step.dependOn(&exe.step);
+    }
+
+我们在这里添加了几行。首先，const exe = b.addExecutable 将创建一个新的 LibExeObjStep，将 src/main.zig 编译成一个名为 fresh 的文件（或 Windows 上的 fresh.exe）。
+
+第二个添加的内容是 compile_step.dependOn(&exe.step);。这就是我们构建依赖关系图的方法，并声明当编译\_step 生成时，exe 也需要生成。
```
Comment (jiacai2050) [inline content/post/2023-12-24-zig-build-explained-part1.md:111]:
```suggestion
第二个添加的内容是 compile_step.dependOn(&exe.step);。这就是我们构建依赖关系图的方法，并声明当执行 `compile_step` 时，`exe` 步骤也需要执行。
```

File: content/post/2023-12-24-zig-build-explained-part1.md [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+    zig build -Doptimize=ReleaseSafe
+    zig build -Doptimize=ReleaseSmall
+
+可以看到，对于布尔选项，我们可以省略 =true，直接设置选项本身。
+
+但我们仍然必须调用 zig build 编译，因为默认调用仍然没有任何作用。让我们改变一下！
+
+## 安装工件
+
+要安装任何东西，我们必须让它依赖于构建器的安装步骤。该步骤是已创建的，可通过 Builder.getInstallStep() 访问。我们还需要创建一个新的 InstallArtifactStep，将我们的 exe 工件复制到安装目录（通常是 zig-out）
```
Comment (jiacai2050) [inline content/post/2023-12-24-zig-build-explained-part1.md:182]:
```suggestion
要安装任何东西，我们必须让它依赖于构建器的安装步骤。该步骤是已创建的，可通过 Builder.getInstallStep() 访问。我们还需要创建一个新的 InstallArtifactStep，将我们的 exe 文件复制到安装目录（通常是 zig-out）
```

File: content/post/2023-12-24-zig-build-explained-part1.md [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+3. 当我们调用 zig build install（或简称 zig build）时，它将创建 InstallArtifactStep。
+4. InstallArtifactStep 会将 exe 的输出文件注册到一个列表中，以便再次卸载它
+
+现在，当你调用 zig build 时，你会看到一个新的目录 zig-out 被创建了.看起来有点像这样：
+
+    zig-out
+    └── bin
+        └── fresh
+
+现在运行 ./zig-out/bin/fresh，就能看到这条漂亮的信息：
```
Comment (jiacai2050) [inline content/post/2023-12-24-zig-build-explained-part1.md:213]:
```suggestion
现在运行 ./zig-out/bin/fresh，就能看到这条信息：
```

File: content/post/2023-12-24-zig-build-explained-part1.md [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+    step-name                   This is what is shown in help
+
+    General Options:
+    ...
+
+请注意，除了在 zig build --help 中添加一个小条目并允许我们调用 zig build step-name 之外，这个步骤仍然没有任何作用。
+
+Step 遵循与 std.mem.Allocator 相同的接口模式，需要实现一个 make 函数。步骤创建时将调用该函数。对于我们在这里创建的步骤，该函数什么也不做。
+
+现在，我们需要创建一个漂亮的 Zig 程序：
```
Comment (jiacai2050) [inline content/post/2023-12-24-zig-build-explained-part1.md:92]:
```suggestion
现在，我们需要创建一个稍正式的 Zig 程序：
```

File: content/post/2023-12-28-zig-build-explained-part2.md [hunk 5]
```diff
@@ -0,0 +1,530 @@
...
+date: "2023-12-24T19:15:02+0800"
+---
+
+# zig build explained - part2
+
+原文链接： https://zig.news/xq/zig-build-explained-part-2-1850
+
+## 注释
+
+从现在起，我将只提供一个最小的 build.zig，说明解决一个问题所需的文件。如果你想了解如何将所有这些文件粘合到一个漂亮舒适的构建文件中，请阅读第一篇文章。
```
Comment (jiacai2050) [inline content/post/2023-12-28-zig-build-explained-part2.md:13]:
```suggestion
从现在起，我将只提供一个最小的 build.zig，说明解决一个问题所需的文件。如果你想了解如何将所有这些文件粘合到一个构建文件中，请阅读第一篇文章。
```

File: content/post/2023-12-28-zig-build-explained-part2.md [hunk 6]
```diff
@@ -0,0 +1,530 @@
...
+
+如你所见，只需向 -target 传递目标三元组，就能调用交叉编译。只需确保所有外部库都已准备好进行交叉编译即可！
+
+## 使用 zig build-exe 和其他工具
+
+使用 Zig 工具链构建 C 项目的另一种方法与构建 Zig 项目的方法相同：
+
+    zig build-exe -lc main.c buffer.c
+
+这里的主要区别在于，必须明确传递 -lc 才能链接到 libc，而且可执行文件的名称将从传递的第一个文件中导出。如果想使用不同的可执行文件名，可通过 --name example 再次获取示例文件。
```
Comment (jiacai2050) [inline content/post/2023-12-28-zig-build-explained-part2.md:49]:
```suggestion
这里的主要区别在于，必须明确传递 -lc 才能链接到 libc，而且可执行文件的名称将从传递的第一个文件中推导出。如果想使用不同的可执行文件名，可通过 --name example 再次获取示例文件。
```

File: content/post/2023-12-28-zig-build-explained-part2.md [hunk 7]
```diff
@@ -0,0 +1,530 @@
...
+
+交叉编译也是如此，只需通过 -target x86_64-windows-gnu 或其他目标三元组即可：
+
+    zig build-exe -lc -target x86_64-windows-gnu main.c buffer.c
+
+你会发现，使用这条编译命令，Zig 会自动在输出文件中附加 .exe 扩展名，并生成 .pdb 调试数据库。如果你在此处传递 --name example，输出文件也会有正确的 .exe 扩展名，所以你不必考虑这个问题。
+
+## 用 build.zig 创建 C 代码
+
+那么，我们如何用 build.zig 构建我们的双文件小范例呢？
```
Comment (jiacai2050) [inline content/post/2023-12-28-zig-build-explained-part2.md:59]:
```suggestion
那么，我们如何用 build.zig 来构建上面的两个示例呢？
```

File: content/post/2023-12-28-zig-build-explained-part2.md [hunk 8]
```diff
@@ -0,0 +1,530 @@
...
+然后，我们通过 addCSourceFile 添加两个 C 语言文件：
+
+    exe.addCSourceFile(.{ .file = std.build.LazyPath.relative("main.c"), .flags = &.{} });
+    exe.addCSourceFile(.{ .file = std.build.LazyPath.relative("buffer.c"), .flags = &.{} });
+
+第一个参数 addCSourceFile 是要添加的 C 或 C++ 文件的名称，第二个参数是该文件要使用的命令行选项列表。
+
+请注意，我们向 addExecutable 传递的是空值，因为我们没有要编译的 Zig 源文件。
+
+现在，调用 zig build 可以正常运行，并在 zig-out/bin 中生成一个漂亮的小可执行文件。很好，我们用 Zig 构建了第一个小 C 项目！
```
Comment (jiacai2050) [inline content/post/2023-12-28-zig-build-explained-part2.md:98]:
```suggestion
现在，调用 zig build 可以正常运行，并在 zig-out/bin 中生成一个可执行文件。很好，我们用 Zig 构建了第一个小 C 项目！
```

File: content/post/2023-12-28-zig-build-explained-part2.md [hunk 9]
```diff
@@ -0,0 +1,530 @@
...
+        if (b.args) |args| {
+            run_cmd.addArgs(args);
+        }
+        const run_step = b.step("run", "Run the app");
+        run_step.dependOn(&run_cmd.step);
+    }
+
+## 条件编译
+
+与 Zig 相比，C 和 C++ 的条件编译方式非常繁琐。由于缺乏 "懒评估"，有时必须根据目标文件来包含/排除文件。你还必须提供宏定义来启用/禁用某些项目功能。
```
Comment (jiacai2050) [inline content/post/2023-12-28-zig-build-explained-part2.md:302]:
```suggestion
与 Zig 相比，C 和 C++ 的条件编译方式非常繁琐。由于缺乏惰性求值的功能，有时必须根据目标环境来包含/排除文件。你还必须提供宏定义来启用/禁用某些项目功能。
```

File: content/post/2023-12-29-zig-build-explained-part3.md [hunk 10]
```diff
@@ -0,0 +1,484 @@
...
+author: Reco
+date: "2023-12-29T19:15:02+0800"
+---
+
+# zig build explained - part3
+
+原文链接： https://zig.news/xq/zig-build-explained-part-3-1ima
+
+注释
+从现在起，我将只提供一个最小的 build.zig，说明解决一个问题所需的文件。如果你想了解如何将所有这些文件粘合到一个漂亮舒适的构建文件中，请阅读第一篇文章。
```
Comment (jiacai2050) [inline content/post/2023-12-29-zig-build-explained-part3.md:12]:
```suggestion

> 从现在起，我将只提供一个最精简的的 build.zig，用来说明解决一个问题所需的步骤。如果你想了解如何将所有这些文件粘合到一个构建文件中，请阅读本系列第一篇文章。
```

File: content/post/2023-12-29-zig-build-explained-part3.md [hunk 11]
```diff
@@ -0,0 +1,484 @@
...
+
+复合项目
```
Comment (jiacai2050) [inline content/post/2023-12-29-zig-build-explained-part3.md:14]:
```suggestion
## 复合项目
```

File: content/post/2023-12-29-zig-build-explained-part3.md [hunk 12]
```diff
@@ -0,0 +1,484 @@
...
+
+如果你想知道为什么我们不为 deploy_tool 和 test_suite 设置目标：
+两者都是为了在主机平台上运行，而不是在目标机器上。
+此外，deploy_tool 还设置了固定的编译模式，因为我们希望快速编译，即使我们编译的是应用程序的调试版本。
+
+## 总结
+
+看完这一大堆文字，你现在应该可以构建任何你想要的项目了。我们已经学会了如何编译 Zig 应用程序，如何为其添加任何类型的外部库，甚至如何为发布管理对应用程序进行后处理。
+
+我们还可以通过少量的工作来构建 C 和 C++ 项目，并将它们部署到各个地方。
```
Comment (jiacai2050) [inline content/post/2023-12-29-zig-build-explained-part3.md:478]:
```suggestion
我们还可以通过少量的工作来构建 C 和 C++ 项目，并将它们部署到各个地方，而不仅仅是 Zig 项目。
```

**Comment by jiacai2050** [comment]
十分感谢！

**Pull request merged**